client/src/main/scala/io/delta/sharing/spark/RemoteDeltaLog.scala

```diff
@@ -227,11 +228,23 @@ class RemoteSnapshot(
   }
 
   private def checkSchemaNotChange(newMetadata: Metadata): Unit = {
-    if (newMetadata.schemaString != metadata.schemaString ||
+    val schemaChangedException = new SparkException(
+      s"""The schema or partition columns of your Delta table has changed since your
+         |DataFrame was created. Please redefine your DataFrame""".stripMargin)
+
+    if (ConfUtils.structuralSchemaMatchingEnabled(spark.sessionState.conf)) {
+      val newSchema = DataType.fromJson(newMetadata.schemaString).asInstanceOf[StructType]
+      val currentSchema = DataType.fromJson(metadata.schemaString).asInstanceOf[StructType]
+
+      if (
+        metadata.partitionColumns != newMetadata.partitionColumns ||
+          !SchemaUtils.isReadCompatible(currentSchema, newSchema)
```

Comment:
Though, reading the comment of the function and also the usage in DeltaSharingSource, it seems used with (newSchema, currentSchema)?

Reply:
I only reversed it because I was looking at the implementation but I misread the subsetting. Changed.

Reply:
Actually my tests started failing after making that change...I think there is an error in the current implementation.